Tasks/AzureResourceManagerTemplateDeploymentV3/models/DeployAzureRG.ts

```diff
@@ -0,0 +1,94 @@
+import tl = require("azure-pipelines-task-lib/task");
+import msRestAzure = require('azure-arm-rest-v2/azure-arm-common');
+import { AzureRMEndpoint } from 'azure-arm-rest-v2/azure-arm-endpoint';
+
+export class AzureRGTaskParameters {
+
+    public action: string;
+    public resourceGroupName: string;
+    public location: string;
+    public csmFile: string;
+    public csmParametersFile: string;
+    public templateLocation: string;
+    public csmFileLink: string;
+    public csmParametersFileLink: string;
+    public overrideParameters: string;
+    public outputVariable: string;
+    public subscriptionId: string;
+    public endpointPortalUrl: string;
+    public deploymentName: string;
+    public deploymentMode: string;
+    public credentials: msRestAzure.ApplicationTokenCredentials;
+    public deploymentOutputs: string;
+    public addSpnToEnvironment: boolean;
+    public connectedService: string;
+    public deploymentScope: string;
+    public managementGroupId: string;
+
+    private async getARMCredentials(connectedService: string): Promise<msRestAzure.ApplicationTokenCredentials> {
+        var azureEndpoint = await new AzureRMEndpoint(connectedService).getEndpoint();
+        return azureEndpoint.applicationTokenCredentials;
+    }
+
+    public async getAzureRGTaskParameters() : Promise<AzureRGTaskParameters>
+    {
+        try {
+
+            //Deployment Scope
+            this.deploymentScope = tl.getInput("deploymentScope");
+            if(!this.deploymentScope){
+                this.deploymentScope = "Resource Group";
+            }
+
+            this.connectedService = tl.getInput("ConnectedServiceName", true);
+            var endpointTelemetry = '{"endpointId":"' + this.connectedService + '"}';
+            console.log("##vso[telemetry.publish area=TaskEndpointId;feature=AzureResourceManagerTemplateDeployment]" + endpointTelemetry);
+            this.endpointPortalUrl = tl.getEndpointDataParameter(this.connectedService, "armManagementPortalUrl", true);
+            this.action = tl.getInput("action");
+
+            //Management Group Id
+            if(this.deploymentScope === "Management Group"){
+                this.managementGroupId = tl.getEndpointDataParameter(this.connectedService, "ManagementGroupId", false);
+            }
+
+            //Subscripion Id
+            this.subscriptionId = tl.getInput("subscriptionName");
+            if(!this.subscriptionId && this.deploymentScope != "Management Group") {
```

Comment:
```suggestion
            if(!this.subscriptionId && (this.deploymentScope === "Subscription" || this.deploymentScope === "Resource Group") {
```

This will ensure that we need not change this logic when we do tenant level support